source/modifyPath.js

```diff
@@ -0,0 +1,55 @@
+import _curry3 from './internal/_curry3.js';
+import _isArray from './internal/_isArray.js';
+import _isObject from './internal/_isObject.js';
+import _has from './internal/_has.js';
+import _assoc from './internal/_assoc.js';
+import _modify from './internal/_modify.js';
+
+
+/**
+ * Creates a shallow clone of the passed object by applying an `fn` function
+ * to the value at the given path.
+ *
+ * The function will not be invoked, and the object will not change
+ * if its corresponding path does not exist in the object.
+ * All non-primitive properties are copied to the new object by reference.
+ *
+ * @func
+ * @memberOf R
+ * @category Object
+ * @sig [Idx] -> (v -> v) -> {k: v} -> {k: v}
+ * @param {Array} path The path to be modified.
+ * @param {Function} fn The function to apply to the path.
+ * @param {Object} object The object to be transformed.
+ * @return {Object} The transformed object.
+ * @example
+ *
+ *      const person = {name: 'James', address: { zipCode: '90216' }};
+ *      R.modifyPath(['address', 'zipCode'], R.reverse, person); //=> {name: 'James', address: { zipCode: '61209' }}
+ *
+ *      // Can handle arrays too
+ *      const person = {name: 'James', addresses: [{ zipCode: '90216' }]};
+ *      R.modifyPath(['addresses', 0, 'zipCode'], R.reverse, person); //=> {name: 'James', addresses: [{ zipCode: '61209' }]}
+ */
+var modifyPath = _curry3(function modifyPath(path, fn, object) {
+  if (!_isObject(object) && !_isArray(object) || path.length === 0) {
```

Comment:
I'd like to see an extra pair of parentheses here.  Yes, `A && B || C` is specified to be `(A && B) || C`, but I think it's too easy to imagine it as `A && (B || C)`.

Reply:
I feel like I removed it, as my linter marked it as not useful